rfcs/0000-centralized-configuration-management.md

```diff
@@ -0,0 +1,384 @@
+Supported use-cases like renaming or adding new configuration may require one or more updates to the following locations:
+
+1. packer definitions
+2. sensitive ansible variables protected with `ansible-vault`
+3. nonsensitive ansible variables in plain-text
+4. ansible build playbooks
+5. ansible launch playbooks
+6. ansible role variables
+7. ansible role definitions
+8. sensitive terraform input variables stored as tfvars files in `keybase`
+9. nonsensitive terraform input variables passed between Jenkins job stages
+10. terraform templated cloud-init user data definitions
+11. terraform module definitions
+12. nonsensitive Jenkins job definitions stored in Jenkins
+13. nonsensitive Jenkins job definitions stored in various Jenkinsfiles
+14. sensitive Jenkins definitions stored with the Jenkins credential plugin
+
+The ability to supply configuration to each tool should be an enabling feature.
+Instead, the sprawling configuration has historically discouraged us from adopting worthy changes, contributed to operators' inadvertent addition of errors, and represents an unfriendly introduction to new contributors who wish to make what would otherwise be simple updates to BFD configuration.
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+AWS Systems Manager (SSM) Parameter Store is a hierarchical, key-value data store typically used for centrally managed sensitive and nonsensitive configuration data.
+In addition to the AWS Console, the Parameter Store is accessed through the Parameter Store API, and can be even more accessible through various tools, including:
+- the AWS CLI via `aws ssm` sub-command
+- terraform resources and data sources with the aws provider
+- ansible workflows through the `community.aws.aws_ssm_parameter_store` module
+
+### Parameter Formatting
+
+We take advantage of Parameter Store's hierarchical nature by formatting the hierarchies into 4-tuple prefixes, culminating in a named _leaf_ node. Together, the 4-tuple prefix and leaf make up a single parameter: `/${project}/${env}/${group}/${sensitivity}/${leaf}`.
+Each member of the tuple represents a scoped level or tier providing increasingly specific context for BFD services and supporting systems.
+The scopes lend themselves well to crafting simple queries and tailored access policies that target the necessary hierarchies of configuration data only.
+The primary use cases for these parameters include infrastructure-as-code, automated-configuration-management, and release-management strategies in `terraform`, `ansible`, and `jenkins` tooling, respectively.
+This proposed format **does not** apply to other potential Parameter Store use-cases.
+
+### Project Scope
+
+Project scope (`${project}`) is the _parameter name root_. In these hierarchies, this is *practically* an invariant that resolves to our project: `bfd`.
```

Comment:
Question: does paca live within bfd or it's own root?

Reply:
> Question: does paca live within bfd or it's own root?

We could adopt an alternative root for RDA if it makes sense. At this time, the conventional `bfd` root will do.